src/EthernaIndex/Areas/Api/Services/VideosControllerService.cs

```diff
@@ -129,29 +129,46 @@ public async Task DeleteAsync(string id, ClaimsPrincipal userClaims)
             logger.AuthorDeletedVideo(id);
         }
 
-        public async Task<VideoDto> FindByIdAsync(string id)
+        public async Task<VideoDto> FindByIdAsync(string id, ClaimsPrincipal userClaims)
         {
             // Get Video.
             var video = await indexDbContext.Videos.FindOneAsync(v => v.Id == id);
 
             // Get VideoManifest.
             var lastValidManifest = video.LastValidManifest;
 
-            // Get User.
-            var sharedInfo = await sharedDbContext.UsersInfo.FindOneAsync(video.Owner.SharedInfoId);
+            // Get Owner User.
+            var ownerSharedInfo = await sharedDbContext.UsersInfo.FindOneAsync(video.Owner.SharedInfoId);
+
+            // Get Current User.
+            var currentUserSharedInfo = await sharedDbContext.UsersInfo.FindOneAsync(ui => ui.EtherAddress == userClaims.GetEtherAddress());
 
-            return new VideoDto(video, lastValidManifest, sharedInfo);
+            // Get Vote.
+            var vote = await indexDbContext.Votes.TryFindOneAsync(v => v.Video.Id == id &&
+                                                                     v.Owner.SharedInfoId == currentUserSharedInfo.Id);
```

Comment:
use `v.Owner.Id` instead. Id is always present, and remove need to find `currentUserSharedInfo` from db